src/Sai2Model.h

```diff
@@ -755,6 +761,58 @@ class Sai2Model {
 	 */
 	void displayJoints();
 	void displayLinks();
+	
+	/**
+	 * @brief Computes the joint selection matrix for the joints leading up to the link
+	 * 
+	 * @param link_name 
+	 * @return MatrixXd 
+	 */
+	MatrixXd linkDependency(const std::string& link_name);
```

Comment:
For this function and all the functions you added, add unit test cases in tests/Tests_Sai2Model.cpp